### ISSUE TITLE
Color picker: the Escape key should not be used to set a new color

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#50877]
### Description

Splitting this out from https://github.com/WordPress/gutenberg/issues/50876

The escape key is universally used to cancel / abort the current operation / action.

However, when using the color picker with a keyboard, the only way I could find to actually _set_ a new color is by pressing the Escape key. That's less than ideal and counterintuitive for all users.

Ideally, there should be a specific, dedicated button to _set_ the new color and close the color picker.
The Escape key should be reserved for cancelling the current color selection and close the color picker.

### Step-by-step reproduction instructions

- Go to the Site editor.
- Edit the Color palette.
- Use the keyboard, focus one of the color buttons, and press Enter to open the color picker.
- Don't make any change to the color value and press the Escape key.
- The color picker closes. So far so good.
- Open again the color picker.
- Tab through the color picker UI and change the color valye, either by using the sliders or manually entering new values.
- Observe there's no 'Apply' button or any other specific button dedicated to set the new color.
- Press the Escape key.
- Observe the new color is set and the color picker closes.
- Expected: Pressing the Escape key should cancel the new color selection and close the color picker without setting a new color value.
- Actual: a new color value is set.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
